### PR TITLE
Dplyr 1.0.0

### PR DESCRIPTION
**Pull request by dtburk**
The only change of much substance here is to take over some work that was previously done by `dplyr::bind_rows()` and incorporate it into `ipumsr::ipums_bind_rows()`. Specifically, `dplyr::bind_rows()` used to remove incompatible attributes with a warning, but now it throws an error instead, so I added code to `ipumsr::ipums_bind_rows()` that removes all attributes from columns with any incompatible attributes and warns the user.

### CODE REVIEW

File: R/bind_rows.r [hunk 1]
```diff
@@ -40,6 +40,23 @@ ipums_bind_rows <- function(..., .id = NULL) {
     }
   })
 
+  # Remove all attributes for shared columns with incompatible attributes
```
Comment (gergness) [inline R/bind_rows.r]: I think this can be simplified now that you're removing attributes on all columns.

I'd delete lines 35-41 & 51-57 and add `d_list <- purrr::map(d_list, zap_ipums_attributes)` outside of your if statement on 45.

File: R/lbl_helpers.r [hunk 2]
```diff
@@ -399,10 +399,42 @@ as_lbl_function <- function(x, env = caller_env()) {
 
 # Adapted from rlang:::abort_coercion
 abort_coercion_function <- function(x) {
-  x_type <- rlang::friendly_type_of(x)
+  x_type <- friendly_type_of(x)
```
Comment (gergness) [inline R/lbl_helpers.r:402]: Ha, sorry for the bad advice before. This looks good to me.

**Review by gergness — COMMENTED** [review]
Looks good! Let me know if you still want to talk!

File: R/bind_rows.r (hunk 1 above)
Comment (dtburk) [reply R/bind_rows.r]: @gergness Here's my understanding of what the code is doing: 

Lines 35-41 are removing IPUMS attributes (which is theoretically only a subset of all attributes) from columns where all attributes match across all the data.frames to be combined. This had the effect that both non-IPUMS attributes, and IPUMS attributes that didn't match across all the data.frames, would be left alone, and `dplyr::bind_rows()` would decide what to do with them (usually it would remove them with a warning, I think).

Lines 51-57 (which I added) remove all attributes from columns where not all attributes match, in an effort to avoid an error from `dplyr::bind_rows()`, which now will throw an error for non-matching attributes (unless the class of the columns has defined special `vctrs` coercion rules, I think). 

So, it sounds like you're suggesting that we simply remove IPUMS attributes for all columns, regardless of whether they match across data.frames (and continue putting back the ones that match after our call to `dplyr::bind_rows()`). The main difference I see with that approach is that non-IPUMS attributes would be left alone and handled by `dplyr::bind_rows()`, which will most likely throw an error (though not if the attributes come from a class that has defined `vctrs` coercion rules). I can see how that makes sense, in that we shouldn't necessarily remove non-IPUMS attributes that might be handled more intelligently by `dplyr::bind_rows()`. The downside I see is that it could lead to more errors coming from `dplyr::bind_rows()`, but I guess they are justified errors that are not the responsibility of `ipumsr` to handle.

So, in conclusion, I guess I agree with you, I just needed to talk it out :) Am I understanding the rationale of your suggestion?

File: R/lbl_helpers.r (hunk 2 above)
Comment (dtburk) [reply R/lbl_helpers.r:402]: Not bad advice, just incomplete, as it turns out.
